indigo/indigo-extras/src/main/scala/indigoextras/ui/RadioButtonGroup.scala

```diff
@@ -16,134 +16,215 @@ import indigo.shared.scenegraph.Group
   * @param onUnselected The events fired when this button ceases to be selected
   * @param onHoverOver The events fired when this button is hovered over
   * @param onHoverOut The events fired when this button is no longer hovered over
+  * @param state The current state of the radio button i.e., selected, hover, or normal
   */
 final case class RadioButton(
     position: Point,
     onSelected: () => List[GlobalEvent],
     onUnselected: () => List[GlobalEvent],
     onHoverOver: () => List[GlobalEvent],
-    onHoverOut: () => List[GlobalEvent]
+    onHoverOut: () => List[GlobalEvent],
+    state: RadioButtonState
 ) {
-  def withSelectedAction(action: () => List[GlobalEvent]): RadioButton =
-    this.copy(onSelected = action)
+  def withSelectedAction(actions: GlobalEvent*): RadioButton =
+    withSelectedAction(actions.toList)
+  def withSelectedAction(actions: => List[GlobalEvent]): RadioButton =
+    this.copy(onSelected = () => actions)
 
-  def withUnselectedAction(action: () => List[GlobalEvent]): RadioButton =
-    this.copy(onUnselected = action)
+  def withUnselectedAction(actions: GlobalEvent*): RadioButton =
+    withUnselectedAction(actions.toList)
+  def withUnselectedAction(actions: => List[GlobalEvent]): RadioButton =
+    this.copy(onUnselected = () => actions)
 
-  def withHoverOverAction(action: () => List[GlobalEvent]): RadioButton =
-    this.copy(onHoverOver = action)
+  def withHoverOverAction(actions: GlobalEvent*): RadioButton =
+    withHoverOverAction(actions.toList)
+  def withHoverOverAction(actions: => List[GlobalEvent]): RadioButton =
```

Comment:
I realise this is in keeping with the above method... but should this be `withHoverOverActions` (plural) as you're adding more than 1?

Reply:
Fixed - good spot - thanks.